test/functionalTest/cases/2507_metadata_flags_in_notifications/create_and_update_csubs_with_metadata_flags_field.test

```diff
@@ -0,0 +1,326 @@
+# 04. Update subscription with metadataFlags: bar, see error
+# 05. Update subscription without touching metadataFlags
+# 06. Get subscription and see metadataFlags: true
+# 07. Update subscription with metadataFlags: false
+# 08. Get subscription and see no metadataFlags
+#
+
+echo "01. Create subscription with metadataFlags: foo, see error"
+echo "=========================================================="
+payload='
+{
+    "subject": {
+        "entities": [
+            {
+                "id": "E1",
+                "type": "T"
+            }
+        ],
+        "condition": {
+            "attrs": [ ]
+         }
+    },
+    "notification": {
+        "http": {"url": "http://localhost:1234/notify"},
+        "metadataFlags": "foo",
+        "attrs": [ ]
+    },
+    "expires": "2050-04-05T14:00:00.00Z"
+}
+'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create subscription with metadataFlags: true"
+echo "================================================"
+payload='
+{
+    "subject": {
+        "entities": [
+            {
+                "id": "E1",
+                "type": "T"
+            }
+        ],
+        "condition": {
+            "attrs": [ ]
+         }
+    },
+    "notification": {
+        "http": {"url": "http://localhost:1234/notify"},
+        "metadataFlags": true,
+        "attrs": [ ]
+    },
+    "expires": "2050-04-05T14:00:00.00Z"
+}
+'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+SUB_ID=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+
+
+echo "03. Get subscription and see metadataFlags: true"
+echo "================================================"
+orionCurl --url /v2/subscriptions
+echo
+echo
+
+
+echo "04. Update subscription with metadataFlags: bar, see error"
+echo "=========================================================="
+payload='{
+  "notification": {
+    "http": {"url": "http://localhost:1234/notify"},
+    "metadataFlags": "bar",
+    "attrs": [ ]
+  }
+}'
+orionCurl --url /v2/subscriptions/$SUB_ID -X PATCH --payload "$payload"
+echo
+echo
+
+
+echo "05. Update subscription without touching metadataFlags"
+echo "======================================================"
+payload='{
+  "expires": "2060-04-05T14:00:00.00Z"
+}'
+orionCurl --url /v2/subscriptions/$SUB_ID -X PATCH --payload "$payload"
+echo
+echo
+
+
+echo "06. Get subscription and see metadataFlags: true"
+echo "================================================"
+orionCurl --url /v2/subscriptions
+echo
+echo
+
+
+echo "07. Update subscription with metadataFlags: false"
+echo "================================================="
+payload='{
+  "notification": {
+    "http": {"url": "http://localhost:1234/notify"},
+    "metadataFlags": false,
+    "attrs": [ ]
+  }
+}'
+orionCurl --url /v2/subscriptions/$SUB_ID -X PATCH --payload "$payload"
+echo
+echo
+
+
+echo "08. Get subscription and see metadataFlags: "
```

Comment:
Yeah, and adding the important word `no`.
Simply, the TOC line differs from the line used in the echo, and the TOC line seems more accurate.